### PR TITLE
Add participants to team on email verification

### PR DESCRIPTION
**Pull request by IvanObreshkov**
## Refactored the whole logic for the fifth time. 
* Now teams are created only when the participant verifies their email
* Random participants are added to existing random teams on email verification
* There are no more admins of random teams as it was confusing.
* The logic for creating a participant via an invite link stays the same.
* Changed method for the verify endpoint from `GET `to `POST`

### CODE REVIEW

File: packages/py-api/py_api/controllers/hackathon/verification_controller.py [hunk 1]
```diff
@@ -1,88 +1,178 @@
...
-            ),
+            ), session=session,
+        )
+        return verified_participant, None
+
+    @classmethod
+    @handle_database_session_transaction
+    def verify_participants(cls, jwt_token: str, session: ClientSession) -> JSONResponse:
+        result_of_decoding = JWTFunctionality.decode_token(jwt_token)
+        if isinstance(result_of_decoding, dict):
```
Comment (asynchroza) [inline packages/py-api/py_api/controllers/hackathon/verification_controller.py:48]: Something yolo is happening here

File: packages/py-api/py_api/controllers/hackathon/verification_controller.py [hunk 2]
```diff
@@ -1,88 +1,178 @@
...
+                        decoded_token, background_tasks, jwt_token,
+                    )
+
+            # Create a new random team if needed
+            return cls.create_team_and_send_email(
+                decoded_token, session, background_tasks, verified_participant, True, jwt_token,
+            )
+
+        if decoded_token.get("team_name") and decoded_token.get("random_participant") is False:
+            # A confirmation email is send to the admin along with the invite link
```
Comment (asynchroza) [inline packages/py-api/py_api/controllers/hackathon/verification_controller.py:82]: Is sent*

File: packages/py-api/py_api/controllers/hackathon/verification_controller.py [hunk 3]
```diff
@@ -1,88 +1,178 @@
...
+                        verified_participant, team, session,
+                        decoded_token, background_tasks, jwt_token,
+                    )
+
+            # Create a new random team if needed
+            return cls.create_team_and_send_email(
+                decoded_token, session, background_tasks, verified_participant, True, jwt_token,
+            )
+
+        if decoded_token.get("team_name") and decoded_token.get("random_participant") is False:
```
Comment (asynchroza) [inline packages/py-api/py_api/controllers/hackathon/verification_controller.py:81]: There's no return statement for the else case?

**Review by asynchroza — COMMENTED** [review]
At this point I can barely remember what these functions do and what's the workflow so my input won't be of any help here. I'm just leaving some nits here.

File: packages/py-api/py_api/controllers/hackathon/verification_controller.py (hunk 1 above)
Comment (IvanObreshkov) [reply packages/py-api/py_api/controllers/hackathon/verification_controller.py:48]: ?

**Review by Aleks1908 — APPROVED** [review]
Mai raboti

**Pull request merged**